app/controllers/submissions_controller.rb

```diff
@@ -800,4 +822,22 @@ def get_all_file_data(revision, grouping, path)
   def parent_params
     params[:grouping_id].nil? ? super : [*super, :grouping_id]
   end
+
+  # Helper that extracts the URL from a url file to send to a user.
+  # If no URL is found, returns a string saying the file cannot be displayed.
+  def extract_url(file_content)
+    file_pattern = /^\[InternetShortcut\]\nURL=(?<url>\S+)/
```

Comment:
This regex was based on after testing a .url file on Windows 10. The structure was this strict. One Drive appears to be too relaxed with the structure and only checks for a url. 
Personally, I don't think file structure matters so long as a URL is somewhere.
I can switch it to a .markusurl and just extract any url that is found.

Reply:
I guess my question is. Does windows treat this as a valid .url file:

```
[InternetShortcut]
otherkey=othervalue
URL=example.com
```

Because the regex won't handle that even. 

Either way, let's go with a .markusurl file but just to be clear, we want to be stricter than "extract any url that is found", we want the file to _only_ contain a url. The only content in the file should be a url (strip whitespace if necessary). If the file contains anything else, your `is_valid_url?` method should return false